rust_snuba/rust_arroyo/src/backends/kafka/mod.rs

```diff
@@ -222,6 +237,18 @@ impl<C: AssignmentCallbacks> ConsumerContext for CustomContext<C> {
                 .unwrap();
             }
 
+            // specifically in rust-rdkafka (not in python confluent-kafka) we are observing a
+            // weird bug where assigning an offset after rebalancing does not work until the
+            // consumer is being polled.
+            //
+            // the poll() may return an offset-related error or advance the internal rdkafka
+            // buffer. but it shouldn't matter which case we hit, because even if poll() returns a
+            // message here, we are resetting offsets to something else, and (hopefully) not commit
+            // anything before that
+            if let Some(Err(_)) = base_consumer.poll(Some(Duration::from_millis(10))) {
+                tracing::info!("polling failed during rebalancing, resetting offsets manually");
```

Comment:
What does "resetting offsets manually" refer to here?

Reply:
i honestly just put that log message here because rdkafka logs some errors on its own, which are to be discarded. the "manually" just means that it's arroyo resetting the offsets with consumer.assign instead of that happening "automatically" within rdkafka